### PR TITLE
clean up ForwardMode and ReverseMode tests

### PR DESCRIPTION
**Pull request by tthsqe12**

### CODE REVIEW

File: enzyme/test/Integration/ReverseMode/sret.cpp [hunk 1]
```diff
@@ -34,5 +34,5 @@ int main() {
     dmyfunction(3,4,5,res);
     APPROX_EQ(res[0], 4*5., 1e-7);
     APPROX_EQ(res[1], 3*5., 1e-7);
-    APPROX_EQ(res[1], 3*4., 1e-7);
```
Comment (wsmoses) [inline enzyme/test/Integration/ReverseMode/sret.cpp:37]: Oh this is exceptionally weird, how was this passing before?

File: enzyme/test/Integration/ForwardMode/fwdandrev.cpp [hunk 2]
```diff
@@ -1,11 +1,11 @@
-// RUN: %clang++ -std=c++11 -fno-exceptions -O0 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -S | %lli -
-// RUN: %clang++ -std=c++11 -fno-exceptions -O1 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -S | %lli -
-// RUN: %clang++ -std=c++11 -fno-exceptions -O2 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -S | %lli -
-// RUN: %clang++ -std=c++11 -fno-exceptions -O3 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -S | %lli -
-// RUN: %clang++ -std=c++11 -fno-exceptions -O0 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -enzyme-inline=1 -S | %lli -
-// RUN: %clang++ -std=c++11 -fno-exceptions -O1 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -enzyme-inline=1 -S | %lli -
-// RUN: %clang++ -std=c++11 -fno-exceptions -O2 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -enzyme-inline=1 -S | %lli -
-// RUN: %clang++ -std=c++11 -fno-exceptions -O3 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -enzyme-inline=1 -S | %lli -
+// RUN: %clang++ -std=c++14 -fno-exceptions -O0 %s -S -emit-llvm -o - | %opt - %OPloadEnzyme %enzyme -S | %lli -
```
Comment (wsmoses) [inline enzyme/test/Integration/ForwardMode/fwdandrev.cpp:1]: Is the c++ version required?

File: enzyme/test/Integration/ReverseMode/sret.cpp (hunk 1 above)
Comment (tthsqe12) [reply enzyme/test/Integration/ReverseMode/sret.cpp:37]: It wasn't passing.

File: enzyme/test/Integration/ForwardMode/fwdandrev.cpp (hunk 2 above)
Comment (tthsqe12) [reply enzyme/test/Integration/ForwardMode/fwdandrev.cpp:1]: something about declaring a variable in a constexpr function being a c++14 feature.

**Pull request merged**